src/features/navigator/NarrativeControl/Share.tsx

```diff
@@ -33,17 +38,24 @@ const permissionValues: Record<UserPermission, SelectOption> = {
 
 const UserPermissionSelection: FC<{
   initialPerm?: UserPermission;
-  onChange: (opts: SelectOption[]) => Promise<void> | void;
+  onChange: (opts: SelectOption[]) => void;
   username: string;
   wsId: number;
 }> = ({ initialPerm, onChange, username, wsId }) => {
+  /* hooks */
+  const syncd = useAppSelector(synchronized);
+
+  /* derived values */
   const permissionOptions = [
     permissionValues.r,
     permissionValues.w,
     permissionValues.a,
   ];
+
+  /* <UserPermissionSelection /> component */
```

Comment:
what is this comment? Is this a planned feature? Or something we chose not to implement?

Reply:
It's just supposed to demarcate the "getting ready to render" part of the component from the actual render biz. There are a few other comments with a similar purpose lying around because these components are somewhat complex.